src/main/java/org/broadinstitute/hellbender/tools/copynumber/formats/collections/LocatableCollection.java

```diff
@@ -0,0 +1,89 @@
+
+import com.google.common.collect.Ordering;
+import htsjdk.samtools.util.Locatable;
+import htsjdk.samtools.util.OverlapDetector;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+
+import java.io.File;
+import java.util.Comparator;
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Represents an immutable, coordinate-sorted (with no overlaps allowed) collection of records
+ * that extend {@link Locatable} (although contigs are assumed to be non-null when writing to file),
+ * a set of mandatory column headers given by a {@link TableColumnCollection},
+ * and lambdas for reading and writing records. Records are sorted using
+ * {@link IntervalUtils#LEXICOGRAPHICAL_ORDER_COMPARATOR}. See the unit test
+ * for a simple example of a subclass.
+ *
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ */
+public abstract class LocatableCollection<RECORD extends Locatable> extends RecordCollection<RECORD> {
+    public static final Comparator<Locatable> LEXICOGRAPHICAL_ORDER_COMPARATOR = IntervalUtils.LEXICOGRAPHICAL_ORDER_COMPARATOR;
+
+    /**
+     * Records are sorted using {@code LEXICOGRAPHICAL_ORDER_COMPARATOR}.
+     */
+    public LocatableCollection(final List<RECORD> records,
+                               final TableColumnCollection mandatoryColumns,
+                               final Function<DataLine, RECORD> recordFromDataLineDecoder,
+                               final BiConsumer<RECORD, DataLine> recordToDataLineEncoder) {
+        super(
+                Utils.nonNull(records).stream().sorted(LEXICOGRAPHICAL_ORDER_COMPARATOR).collect(Collectors.toList()),
+                mandatoryColumns,
+                recordFromDataLineDecoder,
+                recordToDataLineEncoder);
+        validateIntervals(getRecords());
+    }
+
+    /**
+     * @throws IllegalArgumentException if records are not sorted using {@code LEXICOGRAPHICAL_ORDER_COMPARATOR}
+     */
+    public LocatableCollection(final File inputFile,
+                               final TableColumnCollection mandatoryColumns,
+                               final Function<DataLine, RECORD> recordFromDataLineDecoder,
+                               final BiConsumer<RECORD, DataLine> recordToDataLineEncoder) {
+        super(inputFile, mandatoryColumns, recordFromDataLineDecoder, recordToDataLineEncoder);
+        validateIntervals(getRecords());
+    }
+
+    //check for ordering, duplicates, and overlaps
+    private static <T extends Locatable> void validateIntervals(final List<T> records) {
```

Comment:
Can you move the overlap detection (only) to IntervalUtils?  Others (myself included) could use that code.

Reply:
This code is somewhat cribbed from `checkForOverlaps` in `HashedListTargetCollection`.  The naive linear check for overlaps is only valid if the list is sorted, so there's no real way to usefully decouple the checks...